doc/manual/src/design/overview.md

```diff
@@ -2,14 +2,22 @@
 
 Nix consists of layers that operate fairly independently.
 
-At the top is the *command line interface*.
+At the top is the *command line interface*, translating from invocations of Nix executables to interactions with the underlying layers.
 
-Below that is the Nix *expression language*, in which packages and configurations are written.
+Below that is the *Nix language*, a [purely functional programming](https://en.m.wikipedia.org/wiki/Purely_functional_programming) language.
+It is used to compose expressions which ultimately evaluate to self-contained *build plans*, used to derive *build results* from referenced *build inputs*.
```

Comment:
```suggestion
It is used to compose expressions which ultimately evaluate to self-contained *build steps*, used to derive *build results* from referenced *build inputs*.
```

In case we agree to [call derivation graphs "build plans"](https://github.com/NixOS/nix/pull/6420#issuecomment-1105299320).